### PR TITLE
Fixes/workaround for upcoming (> v3.4.3) ggplot2 release

### PR DESCRIPTION
**Pull request by cpsievert**
This is just a draft for now -- starting a PR to get visual testing results back

### TODO

- [x] Double-check these fixes work well with CRAN version of ggplot2

### CODE REVIEW

File: R/ggplotly.R [hunk 1]
```diff
@@ -725,11 +725,13 @@ gg2list <- function(p, width = NULL, height = NULL,
...
-      # determine axis types (note: scale_name may go away someday)
+      # determine axis types (note: scale_name went away in ggplot2 v3.4.3)
       # https://github.com/hadley/ggplot2/issues/1312
-      isDate <- isTRUE(sc$scale_name %in% c("date", "datetime"))
+      isDate <- isTRUE(sc$scale_name %in% c("date", "datetime")) || 
+        inherits(sc, c("ScaleContinuousDatetime", "ScaleContinuousDate"))
       isDateType <- isDynamic && isDate
-      isDiscrete <- identical(sc$scale_name, "position_d")
+      isDiscrete <- identical(sc$scale_name, "position_d") ||
+        inherits(sc, "ScaleDiscretePosition")
```
Comment (cpsievert) [inline R/ggplotly.R:734]: TODO: double-check this is actually a good way to do this

File: R/ggplotly.R [hunk 2]
```diff
@@ -1512,7 +1514,15 @@ scales_add_missing <- function(plot, aesthetics) {
...
 # -------------------------------------------------------------------------
 get_gdefs_ggproto <- function(scales, theme, plot, layers) {
-  guides <- plot$guides$setup(scales)
+  
+  # Proposed change to accomodate
+  # https://github.com/tidyverse/ggplot2/pull/5428
+  # Ensure a 1:1 mapping between aesthetics and scales
+  aesthetics <- lapply(scales, `[[`, "aesthetics")
+  scales     <- rep.int(scales, lengths(aesthetics))
+  aesthetics <- unlist(aesthetics, recursive = FALSE, use.names = FALSE)
```
Comment (cpsievert) [inline R/ggplotly.R:1523]: TODO: is there not a better way to fix this? Seems like there's a leaked abstraction somewhere

**Comment by cpsievert** [comment]
The one remaining failure is due to the fact that the dev version of ggplot2 breaks GGally. I'm just going to ignore that for now since it seems ggplot2 should be fixing that on their end before a release

File: inst/examples/shiny/event_data/tests/shinytest/mytest-expected/001.json
Comment (zeehio) [inline inst/examples/shiny/event_data/tests/shinytest/mytest-expected/001.json:1]: @cpsievert, it seems to me you are hitting https://github.com/jeroen/jsonlite/issues/420 here, that was fixed on jsonlite 1.8.7. Maybe you want to update your jsonlite locally 😉

**Pull request merged**